se_shared/include/se/path_planning/collision_checker.hpp

```diff
@@ -49,81 +49,85 @@ CollisionCheck<T>::CollisionCheck(const VolumeTemplate<T, se::Octree> &volume,
 
 // from voxblox utils planning_utils_inl.h
 
+
 template<typename T>
-bool CollisionCheck<T>::isSphereCollisionFree(const Eigen::Vector3i pos_v) {
+int CollisionCheck<T>::isSphereCollisionFree(const Eigen::Vector3i center) {
+
   int radius_v = static_cast<int>(planning_config_.cand_view_safety_radius / res_); // m/(m/voxel)
+  se::Node<T> *node = nullptr;
+  se::VoxelBlock<T> *block = nullptr;
+  bool is_voxel_block;
+  Eigen::Vector3i prev_pos(0, 0, 0);
   for (int x = -radius_v; x <= radius_v; x++) {
     for (int y = -radius_v; y <= radius_v; y++) {
       for (int z = -radius_v; z <= radius_v; z++) {
         Eigen::Vector3i point_offset_v(x, y, z);
```

Comment:
It might be faster to nest the loops in the opposite order, `z -> y ->x`. Could you test this?

Reply:
it seems to be slower. the other way around z-> y-> x.